geth/jail/jail_test.go

```diff
@@ -196,6 +197,73 @@ func (s *JailTestSuite) TestJailFetch() {
 	<-wait
 }
 
+func (s *JailTestSuite) TestJailLoopInCall() {
+	require := s.Require()
+	require.NotNil(s.jail)
+
+	s.StartTestNode(params.RopstenNetworkID)
+	defer s.StopTestNode()
+
+	// load Status JS and add test command to it
+	s.jail.BaseJS(baseStatusJSCode)
+	s.jail.Parse(testChatID, ``)
+
+	cell, err := s.jail.GetCell(testChatID)
+	require.NoError(err)
+	require.NotNil(cell)
+
+	cellVM := cell.VM()
+	require.NotNil(cellVM)
+
+	cellLoop := cell.Loop()
+	require.NotNil(cellLoop)
+
+	items := make(chan string)
+
+	err = cellVM.Set("__captureResponse", func(val string) otto.Value {
+		items <- val
+
+		return otto.UndefinedValue()
+	})
+	require.NoError(err)
+
+	_, err = cellVM.Run(`
+		function callRunner(namespace){
+			console.log("Initiating callRunner for: ", namespace)
+			return setInterval(function(){
+				__captureResponse(namespace);
+			}, 1000);
+		}
+	`)
+	require.NoError(err)
+
+	// NOTE: Fails to work because it's running intirely
+	_, err = cellVM.Call("callRunner", nil, "softball")
+	require.NoError(err)
+
+	go func() {
+		require.NoError(cellLoop.Run())
```

Comment:
> If you mean adding a test after validating that the items where fully received to validate no item is currently pending in the cellLoop, then I can do that.

This is exactly what I mean. On the other hand, we have a `TODO` comment that we need to handle the loop and any possible errors. So, maybe we can add additional tests when we start fixing it instead of spending time now. 

TLDR; let's leave it as it is for the time being.